include/s3select_oper.h

```diff
@@ -1268,28 +1383,40 @@ class logical_operand : public base_statement
     return std::string("#");//TBD
   }
   virtual value& eval()
-  {bool res;
+  {
+    bool res;
```

Comment:
optimization issue.
in case left side is false , can return false , without evaluate the right side.